### PR TITLE
FALCON-1794:Behaviour mismatch in running Api when scheduled on nativ…

### PR DESCRIPTION
**Pull request by deepakbarr**

### CODE REVIEW

File: scheduler/src/main/java/org/apache/falcon/state/store/jdbc/InstanceBean.java [hunk 1]
```diff
@@ -46,7 +46,7 @@
         @NamedQuery(name = "DELETE_INSTANCE_FOR_ENTITY", query = "delete from InstanceBean a where a.entityId = :entityId"),
         @NamedQuery(name = "UPDATE_INSTANCE", query = "update InstanceBean a set a.cluster = :cluster, a.externalID = :externalID, a.instanceTime = :instanceTime, a.creationTime = :creationTime, a.actualEndTime = :actualEndTime, a.currentState = :currentState, a.actualStartTime = :actualStartTime, a.instanceSequence = :instanceSequence, a.awaitedPredicates = :awaitedPredicates, a.properties = :properties where a.id = :id"),
         @NamedQuery(name = "GET_INSTANCES_FOR_ENTITY_CLUSTER", query = "select OBJECT(a) from InstanceBean a where a.entityId = :entityId AND a.cluster = :cluster"),
-        @NamedQuery(name = "GET_INSTANCES_FOR_ENTITY_CLUSTER_FOR_STATES", query = "select OBJECT(a) from InstanceBean a where a.entityId = :entityId AND a.cluster = :cluster AND a.currentState IN (:currentState)"),
```
Comment (pallavi-rao) [inline scheduler/src/main/java/org/apache/falcon/state/store/jdbc/InstanceBean.java]: Lets not change the DB query itself as it is used in multiple places. For example, it is used to get suspended instances for resuming. In those cases, we want older instances acted upon first. Can move the sorting to the FalconWorkflowEngine.getRunningInstances () instead?

Comment (deepakbarr) [inline scheduler/src/main/java/org/apache/falcon/state/store/jdbc/InstanceBean.java]: Ok. Makes sense. Will update.

Comment (pavankumar526) [inline scheduler/src/main/java/org/apache/falcon/state/store/jdbc/InstanceBean.java]: @deepakbarr Can you update this, we can merge this pull request

**Comment by ajayyadava** [comment]
@deepakbarr This pull request has been lying unresponsive for quite some time. I will be closing this as part of the clean up of pull requests. Your contribution is really welcome and you can rebase and open the pull request any time you want.

**Pull request closed without merging**